### ISSUE TITLE
"Cannot read properties of undefined (reading 'component')" - Error on build with Vue

### ISSUE THREAD

**Issue by TahiraBF** [histoire-dev/histoire#432]
### Describe the bug

When build Histoire docs to serve in a production env - using `npm run story:build` - , we are able to build but at the moment of accessing on the browser and selecting a story, we get a blank page and in the console there is the following error message: `Uncaught TypeError: Cannot read properties of undefined (reading 'component')`.

### Reproduction

- Open reproduction link: https://stackblitz.com/edit/vitejs-vite-x1bm9n
- Run cmd `npm run story:build`
- Run cmd `serve .histoire/dist`
- Select the story `Hello World` from the served view
- Open console to see error

### System Info

```shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 16.14.2 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 7.17.0 - /usr/local/bin/npm
  npmPackages:
    @histoire/plugin-vue: ^0.12.4 => 0.12.4 
    @vitejs/plugin-vue: ^4.0.0 => 4.0.0 
    histoire: ^0.12.4 => 0.12.4 
    vite: ^4.0.4 => 4.0.4
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/histoire-dev/histoire/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/histoire-dev/histoire/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://histoire.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/histoire-dev/histoire/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/histoire-dev/histoire/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by alvarosabu** [comment]
Thanks God somebody reported this, I was working on the reproduction, and I'm also blocked by the exact same error when copy the build assets to AWS

@Akryum I think is a validation missing when iterating over the files

**Comment by Akryum** [comment]
It's fixed by bd9296e267dfa8678532f10b00ab575f98aa9018
